aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionEndpoint.java

```diff
@@ -512,6 +516,9 @@ public Connection waitForConnectionReady() {
                 return new Connection("Thread interrupted waiting for connection acknowledgement");
             }
             LOG.debug("Current endpoint status: " + endpointStatus.get());
+            if (EndpointStatus.CONNECTION_FAILED.equals(endpointStatus.get())) {
+                return new Connection("Connection failed.");
+            }
```

Comment:
This change ensures that if there is the device has no network connection, the subscription will fail immediately, instead of failing due to timeout awaiting the `start_ack` after 10 seconds.